cmd/auth/login.go

```diff
@@ -176,18 +185,34 @@ func setHostAndAccountId(ctx context.Context, profileName string, persistentAuth
 		return err
 	}
 
-	if persistentAuth.Host == "" {
+	// If [HOST] is provided, set the host to the provided positional argument.
+	if len(args) > 0 && persistentAuth.Host == "" {
+		persistentAuth.Host = args[0]
+	}
+
+	// If neither [HOST] nor --host are provided, and the profile has a host, use it.
+	// Otherwise, prompt the user for a host.
+	if len(args) == 0 && persistentAuth.Host == "" {
 		if len(profiles) > 0 && profiles[0].Host != "" {
 			persistentAuth.Host = profiles[0].Host
 		} else {
-			configureHost(ctx, persistentAuth, args, 0)
+			hostName, err := promptForHost(ctx)
+			if err != nil {
+				return err
+			}
+			persistentAuth.Host = hostName
 		}
 	}
```

Comment:
Both of these can be wrapped in `if persistentAuth.Host == "" {` like it was before, and then use if/else on `len(args)`. Now the two statements are coupled; the second depends on the first executing correctly.

Reply:
Done